File: src/main/java/co/com/isoft/horizon/services/PqrsService.java
package co.com.isoft.horizon.services;

import co.com.isoft.horizon.models.PQRS;
import co.com.isoft.horizon.models.Status;
import co.com.isoft.horizon.models.exceptions.ForbiddenStatusChangeException;
import co.com.isoft.horizon.services.exceptions.ResourceNotFoundException;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;

public interface PqrsService {

  Page<PQRS> getAll(PageRequest pageRequest);

  Page<PQRS> getAllThatMatchesTitle(Pageable pageRequest, String title);

  PQRS save(PQRS pqrs);

  PQRS changeStatus(PQRS pqrs, Status status)
      throws ResourceNotFoundException, ForbiddenStatusChangeException;

  PQRS get(Long id) throws ResourceNotFoundException;
}

File: react/src/services/pqrsService.ts
import axiosInstance from "@/utils/axiosInstance";
import { PQRS } from "@/models/PQRS";
import { Page } from "@/types/Page";

const resourcePath = "/pqrs";

const getAll = async (page?: number, title?: string) => {
  const response = await axiosInstance.get<Page<PQRS>>(resourcePath, {
    params: {
      page: page ?? 0,
      title,
    },
  });
  return response.data;
};

const create = async ({
  title,
  description,
}: {
  title: string;
  description: string;
}) => {
  const response = await axiosInstance.post<PQRS>(resourcePath, {
    title,
    description,
  });

  return response.data;
};

const changeStatus = async (id: number, status: string) => {
  const response = await axiosInstance.patch<PQRS>(
    `${resourcePath}/${id}/status`,
    {
      status,
    }
  );

  return response.data;
};

export default { getAll, create, changeStatus };
